sky/serve/autoscalers.py

```diff
@@ -318,12 +315,48 @@ def get_latest_version_with_min_replicas(
     def select_outdated_replicas_to_scale_down(
             self, replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
+        """Select outdated replicas to scale down."""
+
+        latest_ready_replicas = []
+        old_replicas = []
+        for info in replica_infos:
+            if info.version == self.latest_version:
+                if info.is_ready:
+                    latest_ready_replicas.append(info)
+            else:
+                old_replicas.append(info)
+
+        num_latest_ready_replicas = len(latest_ready_replicas)
```

Comment:
should we move those inside the `if self.update_mode == serve_utils.UpdateMode.ROLLING:` branch? those seems not used in latter code.

Reply:
Actually, probably a TODO (and to discuss), we might want to implement those related functions in the enum class?

Reply:
SGTM